### PR TITLE
Blog Design review

### PR DESCRIPTION
**Pull request by chris-lawton**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1376892243)

### Description of Changes Made

Actions the comments made in the Figma to the blog page including:

- Allow more room for the title on the blog and work pages.
- Align authors with the drop cap in the title
- Introduce (500) medium font weight and update heading styles.
- Ensure the number of authors does not affect the page height.
- Adjust spacing for blog meta to have even top and bottom space.
- Increase the spacing between the end of content and related posts.
- Implement heading spacing mixin for rich text headings.
- Extend title filters on the blog listing page to utilize all available space.
- Ensure blog page title is not in the right hand gutter.
- Refactor listing image and avatar components so that the link surrounds the title only.
- Widen the line before heading when used with filters.
- Adjust border color and reduce font size for tags.
- Rename `image` SCSS file (was missing an underscore), add thick border to image streamfield, and apply bleed to the right edge.
- Include drop cap only for heading one and one-b (this will likely be changed again with the upcoming animated letters work).
- Maintain the drop cap on the related posts title.

### How to Test

- Review blog pages on the site or in the pattern library. 
- Ensure that `listing--avatar` and `listing--image` look as expected after the refactor

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [ ] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] New font variants have not been added
- [ ] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates
- [x] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
 {% endblock %}
 
 {% block content %}
-    <div class="grid grid--spacer-large streamfield">
+    <div class="grid grid--spacer-large streamfield lg:relative">
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html:26]: I don't think the relative and absolute positioning is changing the avatar position - only the top margin is having and effect.

**Review by helenb — COMMENTED** [review]
Thanks @chris-lawton mostly looking good - just a couple of things to check.

File: tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html:26]: Without this the authors that appear in the left hand column don't stay in the second column

<details><summary>Details</summary>
<p>

With:

Screenshot 2024-01-29 at 11 33 06

Without:

Screenshot 2024-01-29 at 11 33 12


</p>
</details>

File: tbx/static_src/sass/components/_author.scss [hunk 2]
```diff
@@ -18,6 +18,13 @@
             margin-bottom: 0;
         }
     }
+
+    .template-blog-page & {
+        @include media-query(large) {
+            position: absolute;
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/static_src/sass/components/_author.scss:24]: Without it a post with multiple authors leaves a space between the tags and date/read time:

<details><summary>Details</summary>
<p>

With: 

Screenshot 2024-01-29 at 11 34 35

Without:

Screenshot 2024-01-29 at 11 34 43


</p>
</details>

File: tbx/static_src/sass/components/_blog-meta.scss [hunk 3]
```diff
@@ -7,7 +7,7 @@
     grid-column: 2 / span 4;
 
     @include media-query(large) {
-        margin: $spacer-medium 0 $spacer-small;
+        margin: $spacer-small 0 $spacer-medium;
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/static_src/sass/components/_blog-meta.scss:10]: Yep. I guess we could add 40px in as a spacing variable if it comes up a lot but I don't think it was included in the spacing scale from the design guide.

File: tbx/static_src/sass/components/_image.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        background-color: var(--color--streamfield-background);
+
+        @include media-query(large) {
+            padding: $spacer-medium;
+        }
+    }
+
+    &__caption,
+    &__attribution {
+        text-align: right;
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/static_src/sass/components/_image.scss:19]: Fixed in https://github.com/torchbox/torchbox.com/pull/68/commits/d67c431a70428f975cb619e1fe29452a626bacf4 - needed to make images 100% width so they extend to the right edge of the screen. It does make portrait images quite tall but should be ok.

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
-    <a class="listing-avatar__link listing__link" href="{{ link }}" aria-label="{{ aria_text|default:title }} by {{ name }}">
-        <h2 class="listing-avatar__title listing__title">{{ title }}</h2>
+    <div class="listing-avatar__inner">
         <div class="listing-avatar__avatar">
             {% include "patterns/atoms/avatar/avatar.html" with lazy_load=True %}
         </div>
-
+        <a class="listing-avatar__link listing__link" href="{{ link }}" aria-label="{{ aria_text|default:title }} by {{ name }}">
+            <h2 class="listing-avatar__title listing__title">
+                {{ title }}
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html]: Fixed in https://github.com/torchbox/torchbox.com/pull/68/commits/8ed7c7fa10016ceb07055b76a0cd1af1708da017

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--image.html [hunk 6]
```diff
@@ -1,39 +1,38 @@
...
 <li class="listing-image listing">
-    <div>
-        {# alt text is empty for consistency with listing--avatar.html where the image is inside the link, and therefore not announced #}
-        <picture>
-            <source media="(max-width: 598px)" srcset="{{  listing_mobile_image.url }} 1x, {{ listing_mobile_image_retina.url }} 2x" />
-            <source media="(min-width: 599px)" srcset="{{ listing_desktop_image.url }} 1x, {{ listing_desktop_image_retina.url }} 2x" />
-            <img src="{{ listing_desktop_image.url }}" alt="" loading="lazy" class="listing-image__image" />
-        </picture>
-    </div>
+    {# alt text is empty for consistency with listing--avatar.html where the image is inside the link, and therefore not announced #}
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--image.html:10]: Good spot. `listing--avatar.html` includes `avatar.html` which has empty alt text so i've updated the comment here to reflect that. Also I think regardless of what `listing--avatar.html` does it makes sense to keep the alt text empty here as the images are decorative.

File: tbx/static_src/sass/components/_author.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/static_src/sass/components/_author.scss:24]: Ah no worries - I was testing with just one author.

File: tbx/static_src/sass/components/_image.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/static_src/sass/components/_image.scss:19]: Maybe add a note to the editor's guide to say that it is best to use landscape images for streamfield image blocks?

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--avatar.html]: Looks good - have suggested a couple of comments to add to the code.

File: tbx/project_styleguide/templates/patterns/molecules/listing/listing--image.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/molecules/listing/listing--image.html]: Great, thanks for checking.

File: tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/project_styleguide/templates/patterns/pages/blog/blog_detail.html:26]: I thought I had replied to this... but anyway, I was testing with just one author - thanks for explaining.

**Review by helenb — APPROVED** [review]

File: tbx/static_src/sass/components/_image.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/static_src/sass/components/_image.scss:19]: Added : )

**Pull request merged**